src/troubleshooting.rst

```diff
@@ -0,0 +1,482 @@
+message.
+
+If your site has configured this platform for you, it's probably (1) or (2),
+check you are able to access the platform and notify the site administrator as
+appropriate.
+
+If you are in the progress of setting up a new platform, it's probably (3).
+You might want to check that you've configured the
+:cylc:conf:`global.cylc[platforms][<platform name>]install target` correctly,
+note that this defaults to the platform name if not explicitly set.
+
+
+``OperationalError: disk I/O error``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This means that something was unable to write to a file when it would expect to
+have been able to.
+
+This error usually occurs if when you have exceeded you filesystem quota.
+
+If a Cylc workflow cannot write to the filesystem, it will shutdown. Once
+you've cleared out enough space for the workflow to continue you should be able
+to safely restart it as you would normally using ``cylc play``, the workflow
+will continue where it left off.
+
+
+``socket.gaierror``
+^^^^^^^^^^^^^^^^^^^
+
+This usually means that a host could not be found on the network. The likely
+cause is DNS configuration.
+
+Cylc is a distributed system so needs to be able to identify the hosts it has
+been configured to use (e.g. the servers where you run Cylc workflows or any
+login nodes you submit jobs to).
+Cylc expects each host to have a unique and stable fully qualified domain name
+(FQDN) and to be identifiable from other hosts on the network using this name.
+
+I.E. If a host identifies itself with an FQDN, then we should be able to look it
+from another host by this FQDN. If we can't, then Cylc can't tell which host is
+which and will not be able to function properly.
+
+If the FQDN of a host is reported differently from different hosts on the
+network, then Cylc commands will likely fail. To fix the issue, ensure that the
+DNS setup is consistent.
+
+Sometimes we do not have control over the platforms we use and it is not
+possible to compel system administrators to address these issues. If this is
+the case, you can fall back to IP address based host identification which may
+work (i.e. use IP address rather than host names, makes logs less human
+readable). As a last resort you can also hard-code the host name for each host.
+
+For more information, see
+:cylc:conf:`global.cylc[scheduler][host self-identification]`.
+
+
+``failed/XCPU``
+^^^^^^^^^^^^^^^
+
+``XCPU`` is the signal that most batch systems will use when a job hits its
+execution time limit.
+
+Use :cylc:conf:`flow.cylc[runtime][<namespace>]execution time limit` to
+increase this limit.
+
+
+``Cannot determine whether workflow is running on <host>``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a Cylc workflow runs, it creates a :term:`contact file` which tells us on
+which host and port it can be contacted.
+
+If the workflow cannot be contacted, Cylc will attempt to check whether the
+process is still running to ensure it hasn't crashed.
```

Comment:
```suggestion
When Cylc runs a workflow, it creates a :term:`contact file` which tells us on
which host and port it can be contacted.

If the scheduler cannot be contacted, Cylc will attempt to check whether the
process is still running to ensure it hasn't crashed.
```
(Just another example of my complaint above - I think "contacting *the workflow*" is a bit too loose.